src/ngraph/runtime/gpu/gpu_backend.cpp

```diff
@@ -139,24 +159,11 @@ bool runtime::gpu::GPU_Backend::call(shared_ptr<Function> func,
     m_context->prepare_runtime_context();
 
     // Device tensors
-    vector<void*> inputs;
-    vector<void*> outputs;
-
-    for (size_t i = 0; i < input_tvs.size(); i++)
-    {
-        shared_ptr<runtime::gpu::GPU_TensorView> tv =
-            static_pointer_cast<runtime::gpu::GPU_TensorView>(input_tvs[i]);
-        inputs.push_back(tv->m_allocated_buffer_pool);
-    }
-    for (size_t i = 0; i < output_tvs.size(); i++)
-    {
-        shared_ptr<runtime::gpu::GPU_TensorView> tv =
-            static_pointer_cast<runtime::gpu::GPU_TensorView>(output_tvs[i]);
-        outputs.push_back(tv->m_allocated_buffer_pool);
-    }
+    initialize_io(instance.m_inputs.data(), inputs);
```

Comment:
I guess this just overwrites the cached inputs/outputs every time, correct?

Reply:
should be in this?
```
if (instance.m_external_function == nullptr)
```

Reply:
Yes. The whole point of this is to not have to create new vectors with each call. The vectors are really just used as safe ways to deal with raw memory buffers.